core/src/main/scala/org/apache/spark/network/BlockDataManager.scala

```diff
@@ -30,7 +30,7 @@ trait BlockDataManager {
    * Interface to get local block data. Throws an exception if the block cannot be found or
    * cannot be read successfully.
    */
-  def getBlockData(blockId: BlockId): ManagedBuffer
+  def getBlockData(blockId: BlockId, shuffleGenerationId: Int): ManagedBuffer
```

Comment:
hmm, it's a little weird to have a `shuffleGenerationId` parameter in a general `getBlockData` method.

Reply:
Fully agree, actually in the original patch, I add a new function named getShuffleBlockData, which want only to address getting shuffle block data. But I found that for support old version of fetching blocks by `OpenBlocks`, we still keep the logic of fetching shuffle block data which has no shuffle generation id in `getBlockData` function, so I revert back and implement as the current version. Does Wenchen have any suggestion?

Reply:
We can keep `getBlockData` unchanged, and add a new `getShuffleBlockData` method.

For old shuffle protocol, we still call `getBlockData`. Otherwise, call `getShuffleBlockData`.

Reply:
Thanks for the suggestion, done this in 96bab1e.